### PR TITLE
Implement submuncher.

### PR DESCRIPTION
**Pull request by mateusz**

### CODE REVIEW

File: src/SubMuncher.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+namespace SilverStripe\SubMuncher;
+
+use IPTools\Network;
+use IPTools\Range;
+
+class SubMuncher
+{
+    /**
+     * @var string
```
Comment (stojg) [inline src/SubMuncher.php:11]: int?

File: src/SubMuncher.php [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        return $this->leakTotal;
+    }
+
+    /**
+     * @param Network[] $rules
+     * @param $max
+     *
+     * @return string[]
+     */
+    private function consolidateIpVersion(array $rules, $max)
```
Comment (stojg) [inline src/SubMuncher.php:51]: I would be keen to see a test that demonstrates the use and result of the $max parameter

File: src/SubMuncher.php (hunk 1 above)
Comment (mateusz) [reply src/SubMuncher.php:11]: Nope - potential leak size in IPv6 is larger than PHP_INT_MAX.

Comment (stojg) [reply src/SubMuncher.php:11]: oh lol, yes.

File: src/SubMuncher.php (hunk 2 above)
Comment (mateusz) [reply src/SubMuncher.php:51]: There is one already - consolidates to rules to max 1. Would you like another one that consolidates to some other number than 1?

Comment (stojg) [reply src/SubMuncher.php:51]: yes, that would be A+

Comment (mateusz) [reply src/SubMuncher.php:51]: Done!

**Pull request merged**